packages/react-native/Libraries/TurboModule/TurboModuleRegistry.js

```diff
@@ -88,3 +88,19 @@ export function getEnforcing<T: TurboModule>(name: string): T {
   invariant(module != null, message);
   return module;
 }
+
+interface ModuleHolder<T> {
+  module: T | null
+}
+
+export function getLazy<T: TurboModule>(name: string): T {
+  const proxy = new Proxy<ModuleHolder<T>>({ module: null }, {
+    get: (target, property) => {
+      if (target.module == null) {
+        target.module = getEnforcing(name);
+      }
+      return target.module[property];
+    },
+  });
+  return proxy as T;
+}
```

Comment:
@mrousavy, in general, we want to discourage people from using Proxy. It's extremely slow: I've seen ~20x slowdown quoted internally (for Hermes). And, it may preclude some compile-time optimizations we might want to ship to react native the framework. 

So, unfortunately, we cannot ship this addition to the core.

Reply:
Gotcha, makes sense. I didn't know about these slowdowns, I'll keep that in mind when working on react-native-mmkv - thanks!